dianna/methods/kernelshap_image.py

```diff
@@ -123,12 +127,27 @@ def explain(
             self._runner, np.zeros((len(self.labels), n_segments)))
 
         # Temporarily hide warnings, because shap is very spammy
+        # `shap_values` is a matrix of shapley values for each model outputs
         with LoggingContext(level=logging.CRITICAL):
             shap_values = explainer.shap_values(np.ones(
                 (len(self.labels), n_segments)),
                                                 nsamples=nsamples)
 
-        return shap_values, self.image_segments
+        # create heat_maps where shape is (n_classes, *image_segments.shape)
+        n_classes = len(shap_values)
+        heat_maps = np.zeros((n_classes, *self.image_segments.shape))
+
+        # fill the heat_maps with shap values for each class and segment
+        for i, shap_value in enumerate(shap_values):
+            class_heat_map = heat_maps[i]
+            for index in self.image_segments.flat:
+                class_heat_map[self.image_segments == index] = shap_value[0][index - 1]
+            heat_maps[i] = class_heat_map
```

Comment:
You could consider extracting these lines into its own method and maybe even create a test for it.
Both are just suggestions, doing neither of these things at this point is totally fine with me.

Reply:
I understand that the code will be more readable if this code block goes to its own function. However, I think it does not need to be a method of the class as the `heatmaps` or `image_segments` are already returned. Instead of a method, I refactored the code block as a function in this module.